sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -53,19 +53,12 @@ private[hive] object HiveShim {
    * This function in hive-0.13 become private, but we have to do this to work around hive bug
    */
   private def appendReadColumnNames(conf: Configuration, cols: Seq[String]) {
-    val old: String = conf.get(ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR, "")
-    val result: StringBuilder = new StringBuilder(old)
-    var first: Boolean = old.isEmpty
-
-    for (col <- cols) {
-      if (first) {
-        first = false
-      } else {
-        result.append(',')
-      }
-      result.append(col)
-    }
-    conf.set(ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR, result.toString)
+    val key = ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR
+    val value = Option(conf.get(key, null))
+      .map(old => cols.+:(old))
+      .getOrElse(cols)
+      .mkString(",")
```

Comment:
I don't think this is more readable. Also, the previous code is more performant. I wouldn't change this.

Reply:
As for performance:

Current Version:
```
[info] HiveShimSuite:
[info] - appendReadColumns (549 milliseconds)
```

Previous Version:
```
[info] HiveShimSuite:
[info] - appendReadColumns (949 milliseconds)
```

Reply:
For comprehension is just a syntax sugar and is not performant at all.

Reply:
Right, but I don't think it's more readable. It's non-critical path so performance concern is secondary anyway.

Reply:
thanks HyukjinKwon for the kind explaination. we are from an alibaba subsidiary internet company working on resolving real time data calculation based on Peta level data. Currently we have an real time engine which is built on this spark-sql, well, at lease from the sql engine point of view. The throughput is over an qps of 1k and with response latency less than 100ms. Its an none-stop online 24*7 platform serving over ten million active users. If this market scale and platform scale meets the criteria, let me know, we are searching for further cooporations. thanks again. maomao

Reply:
Thanks. It doesn't matter which company it is. All the PRs are equally and reasonably reviewed, and merged per the same criteria.

Reply:
gotcha